src/stores/events/EventsStore.ts

```diff
@@ -161,17 +167,22 @@ export default class EventsStore {
 			const rootNode = this.eventDataStore.targetNodeParents[0];
 
 			if (
-				(isRootEvent(rootNode) || rootNode.isUnknown) &&
+				isRootEvent(rootNode) &&
+				rootNode.eventId !== unknownRoot.eventId &&
 				!this.eventDataStore.rootEventIds.includes(rootNode.eventId)
 			) {
 				rootIds.push(rootNode.eventId);
 			}
 		}
 
-		const rootNodes = sortEventsByTimestamp(
-			rootIds.map(eventId => this.eventDataStore.eventsCache.get(eventId)).filter(isEventNode),
-			'desc',
-		);
+		const rootNodes = [
+			...new Set(
```

Comment:
I thinks it's better not to add duplicate id's then filter it afterwards. Why do we have duplicates there? unknownRoot is added multiple times?

Reply:
Because we have only real ids in `eventDataStore.rootEventIds` we can get several unknown group nodes.